### PR TITLE
LPD-37684 - Issue with Default SKU Staying in Draft State when Creating Liferay Commerce Product with Workflow via Headless API

### PR DESCRIPTION
**Pull request by victorcdp**
@andrea-ale-sbarra Can you review this? Tied to an LPP. Thank you!

Issue: [LPD-37684](https://liferay.atlassian.net/browse/LPD-37684)

Whenever POSTing a product through headless with nested SKUs, if the Single Approver Workflow is enabled for Product, the SKUs will be created in a Draft state instead of Approved

This happens because the Workflow Action is set as SAVE_DRAFT when creating the CPDefinition, so later when updating/creating the nested resources, the SKU will be saved as Draft.

The fix's purpose is to restore the original workflow action before updating/creating the nested resources, and then switch back once that's done. The workflow action of the main object shouldn't interfere in the creation of the nested objects.

### CODE REVIEW

**Comment by victorcdp** [comment]
ci:test:sf

File: modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-catalog-test/src/testIntegration/java/com/liferay/headless/commerce/admin/catalog/resource/v1_0/test/ProductResourceTest.java [hunk 1]
```diff
@@ -674,6 +679,28 @@ private void _testPostProductWithProductChannelExternalReferenceCode()
...
 
+	private void _testPostProductWithWorkflowSingleApprover() throws Exception {
+		_workflowDefinitionLinkLocalService.updateWorkflowDefinitionLink(
+			TestPropsValues.getUserId(), testCompany.getCompanyId(),
+			testCompany.getGroupId(), CPDefinition.class.getName(), 0, 0,
+			"Single Approver@1");
+
+		Product randomProduct = _randomProductWithSku();
+
+		Product postProduct = productResource.postProduct(randomProduct);
```
Comment (alessiorendina) [inline modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-catalog-test/src/testIntegration/java/com/liferay/headless/commerce/admin/catalog/resource/v1_0/test/ProductResourceTest.java:690]: You can inline here

**Review by alessiorendina — CHANGES_REQUESTED** [review]
@victorcdp LGTM, just 1 small change. Thank you!

**Comment by andrea-ale-sbarra** [comment]
ci:forward:force

**Pull request closed without merging**